### PR TITLE
Fix get_dates

### LINKED ISSUE #531: BUG: get_dates returning wrong type?
I tried to run 
```
regolith build recent-collabs --people sbillinge
```
and I got this:
```
loading ..\db\citations.yml...
loading ..\db\contacts.yml...
loading ..\db\people.yml...
loading ../../rg-db-public/db\citations.yml...
loading ../../rg-db-public/db\institutions.yml...
loading ../../rg-db-public/db\people.yml...
Traceback (most recent call last):
  File "C:/Users/simon/Anaconda3/envs/bg/Scripts/regolith", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "C:\Users\simon\dev\regolith\scripts\regolith", line 3, in <module>
    main()
  File "c:\users\simon\dev\regolith\regolith\main.py", line 336, in main
    CONNECTED_COMMANDS[rc.cmd](rc)
  File "c:\users\simon\dev\regolith\regolith\commands.py", line 144, in build
    bldr.build()
  File "c:\users\simon\dev\regolith\regolith\builders\basebuilder.py", line 100, in build
    getattr(self, cmd)()
  File "c:\users\simon\dev\regolith\regolith\builders\coabuilder.py", line 497, in excel
    query_results = self.query_ppl(target, since_date=since_date)
  File "c:\users\simon\dev\regolith\regolith\builders\coabuilder.py", line 415, in query_ppl
    my_collabs = get_coauthors_from_pubs(pubs, person)
  File "c:\users\simon\dev\regolith\regolith\builders\coabuilder.py", line 127, in get_coauthors_from_pubs
    for pub in pubs:
  File "c:\users\simon\dev\regolith\regolith\builders\coabuilder.py", line 105, in filter_since_date
    if has_started(pub, since_date):
  File "c:\users\simon\dev\regolith\regolith\dates.py", line 347, in has_started
    dates = get_dates(thing)
  File "c:\users\simon\dev\regolith\regolith\dates.py", line 242, in get_dates
    date = datetime.date(thing[datenames[2]],
TypeError: an integer is required (got type str)
```

### PR DESCRIPTION
**Pull request by mwinitch**
Closes #531

### CODE REVIEW

**Comment by mwinitch** [comment]
@sbillinge I believe the issue is that sometimes the `thing.get('year)` does return a `str` not an `int`. I just made a simple function called `year_to_int` that converts the year to the int. I could have used the `month_to_int` but I know the code should be Pythonic and I believe this makes it more readable.

File: regolith/dates.py [hunk 1]
```diff
@@ -49,6 +49,14 @@
 }
 
 
+def year_to_int(y):
```
Comment (sbillinge) [inline regolith/dates.py:52]: I see.  we don't need a ``year_to_int`` function, just do ``int(year)`` in the code.

**Comment by mwinitch** [comment]
@sbillinge This is ready for review and merge

**Comment by sbillinge** [comment]
thanks @mwinitch

**Pull request merged**